### PR TITLE
CLS2-1000: Handle EYB leads that have no company gracefully

### PR DESCRIPTION
**Pull request by samuele-mattiuzzo**
## Description of change

Takes care of [CLS2-1000](https://uktrade.atlassian.net/browse/CLS2-1000)

This ticket ensures the frontend handles gracefully when a `eybLead.company` object does not exist (yet).

This is going to be temporary and just a workaround solution.

## Test instructions

- Instead of `eybLead.company.name` we'll instead use `eybLead.companyName` when `company` is not yet synced an available (in ListView and DetailView)

- When a `eybLead.company` is not available, any value for `eybLead.company.*` will be replaced by `Not set`

- When a `eybLead.company` is not available, the breadcrumbs will not use the company name; links will not be rendered; the `Create investment project` button will not be rendered

### After

Screenshot 2024-10-18 at 16 53 57

Screenshot 2024-10-18 at 16 53 47

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [x] Has the branch been rebased to main?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [x] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

[CLS2-1000]: https://uktrade.atlassian.net/browse/CLS2-1000?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by baarkerlounger** [comment]
Should there be some tests here? I'm wary of temporary solutions that live longer than expected... 😅

**Comment by oliverjwroberts** [comment]
> Should there be some tests here? I'm wary of temporary solutions that live longer than expected... 😅

EYB leads will only temporarily be missing a `.company` attribute until they are linked (a separate job will run after they have been ingested to do this). However, the handling of these leads should be permanent. Agree with Dan here, it would be good to have a couple of tests here (one for each page).

**Review by baarkerlounger — APPROVED** [review]

**Comment by samuele-mattiuzzo** [comment]
@oliverjwroberts tested :) can you give it a second pair of eyes please?

File: src/client/modules/Investments/EYBLeads/EYBLeadDetails.jsx [hunk 1]
```diff
@@ -35,11 +42,15 @@ const EYBLeadDetails = () => {
               />
               <SummaryTable.Row
                 heading="Sector or industry"
-                children={eybLead.sector.name}
+                children={eybLead.sector ? eybLead.sector.name : 'Not set'}
```
Comment (marijnkampf) [inline src/client/modules/Investments/EYBLeads/EYBLeadDetails.jsx:45]: There is a NOT_SET_TEXT constant defined for this in src/apps/companies/constants.js

File: src/client/modules/Investments/EYBLeads/EYBLeadDetails.jsx [hunk 2]
```diff
@@ -35,11 +42,15 @@ const EYBLeadDetails = () => {
...
               />
               <SummaryTable.Row
                 heading="Location of company headquarters"
-                children={eybLead.companyLocation.name}
+                children={
+                  eybLead.companyLocation
+                    ? eybLead.companyLocation.name
+                    : 'Not set'
```
Comment (marijnkampf) [inline src/client/modules/Investments/EYBLeads/EYBLeadDetails.jsx:52]: There is a NOT_SET_TEXT constant defined for this in src/apps/companies/constants.js

File: src/client/modules/Investments/EYBLeads/EYBLeadName.jsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { EYBLeadResource } from '../../../components/Resource'
 
 const EYBLeadName = (props) => (
   <EYBLeadResource.Inline {...props}>
-    {(eybLead) => eybLead.company.name}
+    {(eybLead) => (eybLead.company ? eybLead.company.name : 'Not set')}
```
Comment (marijnkampf) [inline src/client/modules/Investments/EYBLeads/EYBLeadName.jsx:7]: There is a NOT_SET_TEXT constant defined for this in src/apps/companies/constants.js

File: test/functional/cypress/specs/investments/eyb-lead-details-spec.js [hunk 4]
```diff
@@ -129,4 +129,82 @@ describe('EYB lead details', () => {
         )
     })
   })
+  context('When viewing an EYB lead without a company associated to it', () => {
+    const eybLeadWithValues = eybLeadFaker({
```
Comment (swenban) [inline test/functional/cypress/specs/investments/eyb-lead-details-spec.js:133]: The tests would be clearer if this const is renamed to something like `eybLeadWithoutLinkedCompanyValues`

**Review by marijnkampf — APPROVED** [review]

**Review by swenban — APPROVED** [review]

**Pull request merged**